src/components/color_picker/color_palette_picker/color_palette_picker.tsx

```diff
@@ -0,0 +1,145 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { FunctionComponent } from 'react';
+
+import { EuiSuperSelect } from '../../form';
+
+import { CommonProps } from '../../common';
+
+import { getLinearGradient, getFixedLinearGradient } from '../utils';
+import { ColorStop } from '../color_stops';
+
+import { EuiSuperSelectProps } from '../../form/super_select';
+
+export interface EuiColorPalettePickerPaletteProps {
+  /**
+   *  For storing unique value of item
+   */
+  value: string;
+  /**
+   *  The name of your palette
+   */
+  title?: string;
+  /**
+   * Specify if the palette is
+   * `fixed`: individual color blocks; or
+   * `gradient`: each color fades into the next; or
+   * `text`: a text only option (a title is required).
+   */
+  type: 'fixed' | 'gradient' | 'text';
+  /**
+   * Array of color `strings` or `ColorStops` in the form of
+   * { stop: number, color: string }
+   */
+  palette?: string[] | ColorStop[];
+}
+
+export type EuiColorPalettePickerProps<T extends string> = CommonProps &
+  Omit<
+    EuiSuperSelectProps<T>,
+    'options' | 'itemLayoutAlign' | 'hasDividers'
+  > & {
+    /**
+     *  Specify what should be displayed after a selection: a `palette` or `title`
+     */
+    selectionDisplay?: 'palette' | 'title';
+
+    /**
+     * An array of #EuiColorPalettePickerPalette objects
+     */
+    palettes: EuiColorPalettePickerPaletteProps[];
+  };
+
+export const EuiColorPalettePicker: FunctionComponent<
+  EuiColorPalettePickerProps<string>
+> = ({
+  className,
+  compressed = false,
+  disabled,
+  fullWidth = false,
+  isInvalid = false,
+  onChange,
+  readOnly = false,
+  valueOfSelected,
+  palettes,
+  append,
+  prepend,
+  selectionDisplay = 'palette',
+  ...rest
+}) => {
+  const getPalette = (palette: [], type: string) => {
+    const background =
+      type === 'fixed'
+        ? getFixedLinearGradient(palette)
+        : getLinearGradient(palette);
+
+    return (
+      <div
+        className="euiColorPalettePicker__itemGradient"
+        style={{ background }}
+      />
+    );
+  };
+
+  const getText = (title: any) => title;
+
+  const getInputDisplay = (title: string, palette: [], type: string) => {
+    if (selectionDisplay === 'title') {
+      return getText(title);
+    } else {
+      return type === 'text' ? getText(title) : getPalette(palette, type);
+    }
+  };
+
+  const paletteOptions = palettes.map((item: any) => {
```

Comment:
```suggestion
  const paletteOptions = palettes.map((item: EuiColorPalettePickerPaletteProps) => {
```

I think this is right? Seems to have that shape